src/Packagist/WebBundle/Entity/PackageRepository.php

```diff
@@ -279,7 +279,7 @@ public function getFilteredQueryBuilder(array $filters = array())
             $qb->leftJoin('v.tags', 't');
         }
 
-        $qb->orderBy('p.id', 'DESC');
```

Comment:
you should order first by abandoned, and then by id (to sort packages being ex-aequo for abandoned). sorting by id first would not let any ex-aequo, making subsequent sorting a no-op

And regarding the controllers overriding the ordering, this is an issue, as you would have to sort by abandoned in the controllers too (as they delete any order set here). a better solution would be to pass an argument to the method to tell it to order by name or no.